opencga-storage/opencga-storage-core/src/main/java/org/opencb/opencga/storage/core/variant/annotation/annotators/extensions/cosmic/CosmicVariantAnnotatorExtensionTask.java

```diff
@@ -0,0 +1,173 @@
+        this.options = options;
+        this.objectReader = JacksonUtils.getDefaultObjectMapper().readerFor(new TypeReference<List<EvidenceEntry>>() {});
+    }
+
+    @Override
+    public List<URI> setup(URI output) throws Exception {
+        // Sanity check
+        Path cosmicFile = Paths.get(options.getString(VariantStorageOptions.ANNOTATOR_EXTENSION_COSMIC_FILE.key()));
+        if (!Files.exists(cosmicFile)) {
+            throw new IllegalArgumentException("COSMIC file " + cosmicFile + " does not exist");
+        }
+        cosmicVersion = (String) options.getOrDefault(VariantStorageOptions.ANNOTATOR_EXTENSION_COSMIC_VERSION.key(), null);
+        if (StringUtils.isEmpty(cosmicVersion)) {
+            throw new IllegalArgumentException("Missing COSMIC version");
+        }
+        assembly = (String) options.getOrDefault(VariantStorageOptions.ASSEMBLY.key(), null);
+        if (StringUtils.isEmpty(assembly)) {
+            throw new IllegalArgumentException("Missing assembly");
+        }
+
+        // Clean and init RocksDB
+        dbLocation = Paths.get(output.getPath()).toAbsolutePath().resolve(COSMIC_ANNOTATOR_INDEX_NAME);
+        if (Files.exists(dbLocation)) {
+            // Skipping setup but init RocksDB
+            logger.info("Skipping setup, it was already done");
+            initRockDB(false);
+        } else {
+            logger.info("Setup and populate RocksDB");
+
+            // Init RocksDB
+            initRockDB(true);
+
+            // Call COSMIC parser
+            try {
+                CosmicExtensionTaskCallback callback = new CosmicExtensionTaskCallback(rdb);
+                CosmicParser.parse(cosmicFile, cosmicVersion, ID, assembly, callback);
+            } catch (IOException e) {
+                throw new ToolException(e);
+            }
+        }
+        return Collections.singletonList(dbLocation.toUri());
+    }
+
+    @Override
+    public void checkAvailable() throws IllegalArgumentException {
+        if (dbLocation == null || !Files.exists(dbLocation)) {
+            throw new IllegalArgumentException("COSMIC annotator extension is not available");
+        }
+    }
+
+    @Override
+    public ObjectMap getOptions() {
+        return options;
+    }
+
+    @Override
+    public ObjectMap getMetadata() {
+        return new ObjectMap("data", ID)
+                .append("version", cosmicVersion)
+                .append("assembly", assembly);
+    }
+
+    @Override
+    public List<VariantAnnotation> apply(List<VariantAnnotation> list) throws Exception {
+        for (VariantAnnotation variantAnnotation : list) {
+            Variant variant;
+            try {
+                variant = new Variant(variantAnnotation.getChromosome(), variantAnnotation.getStart(), variantAnnotation.getReference(),
+                        variantAnnotation.getAlternate());
+            } catch (Exception e) {
+                logger.warn("Skipping variant: " + e.getMessage());
```

Comment:
Why is this? Which exceptions are you expecting here?

Reply:
I got `Unknown end or length of the variant '1:1564802-null::<DEL>', type 'DELETION'`; but I think I should use the variant constructor with the parameter `end`, i.e.:

`            Variant variant = new Variant(variantAnnotation.getChromosome(), variantAnnotation.getStart(), variantAnnotation.getEnd(),
                    variantAnnotation.getReference(), variantAnnotation.getAlternate());
`

Reply:
Indeed! You should use that constructor instead, and remove that try-catch